sdk/scheduler/src/main/java/com/mesosphere/sdk/curator/CuratorStateStore.java

```diff
@@ -193,14 +192,45 @@ public void storeTasks(Collection<Protos.TaskInfo> tasks) throws StateStoreExcep
 
     @Override
     public void storeStatus(Protos.TaskStatus status) throws StateStoreException {
-        String taskName;
+        Optional<String> optionalTaskName = Optional.empty();
+
+        //TODO(MB): keep the commented code below: original approach for getting TaskName
+        /* We used to get TaskName from TaskID's prefix, instead of comparing TaskIDs.
+           TaskID is our unique identifier, can not change, but we can modify TaskName
```

Comment:
Yes of course.  You can change taskName without a launch because it is just a key to stateStore. You still get status of your task with taskID.